packages/light-apps/src/ContextGate/ContextGate.tsx

```diff
@@ -19,9 +21,22 @@ export function ContextGate(props: { children: React.ReactNode }): React.ReactEl
     <AlertsContextProvider>
       <TxQueueContextProvider>
         <ApiContextProvider loading={<Loading active>Connecting to the node...</Loading>}>
-          <KeyringContextProvider>
-            <StakingContextProvider>{children}</StakingContextProvider>
-          </KeyringContextProvider>
+          <ApiContext.Consumer>
+              {({ api, isReady, system }: Partial<ApiContextType>): React.ReactElement | boolean | undefined => {
+                return api && isReady && system && (
+                  <StakingContextProvider>
```

Comment:
I would argue to remove StakingContext if the next PR: since this should be substrate-generic,staking might not be present.

Reply:
fair enough, I don't like it either